### PR TITLE
Move Setup information into setup filed

### PR DESCRIPTION
**Pull request by shashank-elastic**
## Issues
https://github.com/elastic/ia-trade-team/issues/191

## Summary

Move Setup info into the respective setup field. 
This needs to be merged once the issue mentioned in the comment https://github.com/elastic/ia-trade-team/issues/191#issuecomment-1771110361

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by terrancedejesus** [comment]
@shashank-elastic - [This](https://github.com/elastic/detection-rules/blob/9f41c9f35cd568c397c1f81bafc997e92bbf2df0/detection_rules/rule.py#L276) needs to be changed to `setup: Optional[definitions.Markdown] = field(metadata=dict(metadata=dict(min_compat="8.3")))`. During the rule object conversion to Kibana API format, each field:value is [formatted](https://github.com/elastic/detection-rules/blob/9f41c9f35cd568c397c1f81bafc997e92bbf2df0/detection_rules/rule_formatter.py#L54). If the type defined in `BaseRuleData` is `definitions.Markdown` it will preserve the formatting. Since `setup` was defined as `str` the value was formatted and cleaned, thus removing newline characters.

Once changed, you can run `view-rule` CLI command on any of the changed rules with `setup` directly defined and you will see that `\n` is preserved now in the string.

To test via import, run `python -m detection_rules dev build-release --add-historical-data yes`. You can find the `releases/` folder in the root of detection rules. Under `8.12/extras` you will find a `8.12-consolidated-rules.ndjson` file with each rule in API format. CTRL+F for your rule name and remove any rule before and after, also change the rule ID slightly. Then in your stack under Security > Rules, Select Import Rules and point to the consolidated NDJSON file. The rule should import and then you can check the setup guide formatting to see if it was fixed.

Let me know if you have any questions, happy to walk through it.

**Comment by shashank-elastic** [comment]
Post Fixing the Formatting of the setup field.

- view-rule command shows new line characters in setup field. 
```
(venv_detection_rules) shashankks@IN-Shashank detection-rules % python -m detection_rules view-rule rules/linux/command_and_control_cat_network_activity.toml

█▀▀▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄   ▄      █▀▀▄ ▄  ▄ ▄   ▄▄▄ ▄▄▄
█  █ █▄▄  █  █▄▄ █    █   █  █ █ █▀▄ █      █▄▄▀ █  █ █   █▄▄ █▄▄
█▄▄▀ █▄▄  █  █▄▄ █▄▄  █  ▄█▄ █▄█ █ ▀▄█      █ ▀▄ █▄▄█ █▄▄ █▄▄ ▄▄█

{
  "author": [
    "Elastic"
  ],
  "description": "This rule monitors for the execution of the cat command, followed by a connection attempt by the same process. Cat is capable of transfering data via tcp/udp channels by redirecting its read output to a /dev/tcp or /dev/udp channel. This activity is highly suspicious, and should be investigated. Attackers may leverage this capability to transfer tools or files to another host in the network or exfiltrate data while attempting to evade detection in the process.",
  "from": "now-9m",
  "index": [
    "logs-endpoint.events.*"
  ],
  "language": "eql",
  "license": "Elastic License v2",
  "name": "Network Activity Detected via cat",
  "query": "sequence by host.id, process.entity_id with maxspan=1s\n  [process where host.os.type == \"linux\" and event.action == \"exec\" and event.type == \"start\" and \n   process.name == \"cat\"]\n  [network where host.os.type == \"linux\" and event.action in (\"connection_attempted\", \"disconnect_received\") and \n   process.name == \"cat\"]\n",
  "related_integrations": [
    {
      "package": "endpoint",
      "version": "^8.2.0"
    }
  ],
  "required_fields": [
    {
      "ecs": true,
      "name": "event.action",
      "type": "keyword"
    },
    {
      "ecs": true,
      "name": "event.type",
      "type": "keyword"
    },
    {
      "ecs": true,
      "name": "host.id",
      "type": "keyword"
    },
    {
      "ecs": true,
      "name": "host.os.type",
      "type": "keyword"
    },
    {
      "ecs": true,
      "name": "process.entity_id",
      "type": "keyword"
    },
    {
      "ecs": true,
      "name": "process.name",
      "type": "keyword"
    }
  ],
  "risk_score": 47,
  "rule_id": "afd04601-12fc-4149-9b78-9c3f8fe45d39",
  "setup": "\nThis rule requires data coming in from Elastic Defend.\n\n### Elastic Defend Integration Setup\nElastic Defend is integrated into the Elastic Agent using Fleet. Upon configuration, the integration allows\nthe Elastic Agent to monitor events on your host and send data to the Elastic Security app.\n\n#### Prerequisite Requirements:\n- Fleet is required for Elastic Defend.\n- To configure Fleet Server refer to the [documentation](https://www.elastic.co/guide/en/fleet/current/fleet-server.html).\n\n#### The following steps should be executed in order to add the Elastic Defend integration on a Linux System:\n- Go to the Kibana home page and click Add integrations.\n- In the query bar, search for Elastic Defend and select the integration to see more details about it.\n- Click Add Elastic Defend.\n- Configure the integration name and optionally add a description.\n- Select the type of environment you want to protect, either Traditional Endpoints or Cloud Workloads.\n- Select a configuration preset. Each preset comes with different default settings for Elastic Agent, you can further customize these later by configuring the Elastic Defend integration policy. [Helper guide](https://www.elastic.co/guide/en/security/current/configure-endpoint-integration-policy.html).\n- We suggest to select \"Complete EDR (Endpoint Detection and Response)\" as a configuration setting, that provides \"All events; all preventions\"\n- Enter a name for the agent policy in New agent policy name. If other agent policies already exist, you can click the Existing hosts tab and select an existing policy instead.\nFor more details on Elastic Agent configuration settings, refer to the [helper guide](https://www.elastic.co/guide/en/fleet/8.10/agent-policy.html).\n- Click Save and Continue.\n- To complete the integration, select Add Elastic Agent to your hosts and continue to the next section to install the Elastic Agent on your hosts.\nFor more details on Elastic Defend refer to the [helper guide](https://www.elastic.co/guide/en/security/current/install-endpoint.html).\n\n",
  "severity": "medium",
  "tags": [
    "Domain: Endpoint",
    "OS: Linux",
    "Use Case: Threat Detection",
    "Tactic: Command and Control",
    "Data Source: Elastic Defend"
  ],
  "threat": [
    {
      "framework": "MITRE ATT&CK",
      "tactic": {
        "id": "TA0011",
        "name": "Command and Control",
        "reference": "https://attack.mitre.org/tactics/TA0011/"
      },
      "technique": []
    },
    {
      "framework": "MITRE ATT&CK",
      "tactic": {
        "id": "TA0005",
        "name": "Defense Evasion",
        "reference": "https://attack.mitre.org/tactics/TA0005/"
      },
      "technique": []
    },
    {
      "framework": "MITRE ATT&CK",
      "tactic": {
        "id": "TA0010",
        "name": "Exfiltration",
        "reference": "https://attack.mitre.org/tactics/TA0010/"
      },
      "technique": []
    }
  ],
  "type": "eql",
  "version": 2
}
(venv_detection_rules) shashankks@IN-Shashank detection-rules % 
```

**Comment by shashank-elastic** [comment]
Imported the rule into the stack. Setup Guide is as expected.
image

File: rules/linux/command_and_control_cat_network_activity.toml [hunk 1]
```diff
@@ -19,7 +19,9 @@ index = ["logs-endpoint.events.*"]
 language = "eql"
 license = "Elastic License v2"
 name = "Network Activity Detected via cat"
-note = """## Setup
+risk_score = 47
+rule_id = "afd04601-12fc-4149-9b78-9c3f8fe45d39"
+setup = """
```
[the remark this answers is not in the document]
Comment (shashank-elastic) [reply rules/linux/command_and_control_cat_network_activity.toml:24]: The setup information is followed from Line 25 - 50.
The content was already as part of note =" ###Setup" we have moved it its own supported field for 8.3+ as 
setup = "<<content>>".

**Review by Aegrah — APPROVED** [review]

**Review by terrancedejesus — APPROVED** [review]
Looks good to me

**Pull request merged**